src/finch/tensor.py

```diff
@@ -412,11 +416,36 @@ def random(shape, density=0.01, random_state=None):
     return Tensor(jl.fsprand(*args))
 
 
+def asarray(obj, /, *, dtype=None, format=None):
+    if format not in {"coo", "csr", "csc", "csf", "dense", None}:
+        raise ValueError(f"{format} format not supported.")
+    tensor = obj if isinstance(obj, Tensor) else Tensor(obj)
+
+    if format is not None:
+        order = tensor.get_order()
+        if format == "coo":
+            storage = Storage(SparseCOO(tensor.ndim, Element(tensor.fill_value)), order)
+        elif format == "csr":
+            storage = Storage(Dense(SparseList(Element(tensor.fill_value))), order)
+        elif format == "csc":
+            storage = Storage(Dense(SparseList(Element(tensor.fill_value))), order)
+        elif format == "csf":
+            storage = Storage(Dense(SparseList(SparseList(Element(tensor.fill_value)))), order)
```

Comment:
CSF is NDim-agnostic, it's just `Dense` followed by however many `SparseList` we need.

Reply:
Right - updated!